src/ape_test/_cli.py

```diff
@@ -1,9 +1,77 @@
+import subprocess
 import sys
+import threading
+import time
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import List, Sequence
 
 import click
 import pytest
+from watchdog import events  # type: ignore
+from watchdog.observers import Observer  # type: ignore
 
 from ape.cli import ape_cli_context
+from ape.utils import ManagerAccessMixin, cached_property
+
+# Copied from https://github.com/olzhasar/pytest-watcher/blob/master/pytest_watcher/watcher.py
+trigger_lock = threading.Lock()
+trigger = None
+
+
+def emit_trigger():
+    """
+    Emits trigger to run pytest
+    """
+
+    global trigger
+
+    with trigger_lock:
+        trigger = datetime.now()
+
+
+class EventHandler(events.FileSystemEventHandler, ManagerAccessMixin):
+    EVENTS_WATCHED = (
+        events.EVENT_TYPE_CREATED,
+        events.EVENT_TYPE_DELETED,
+        events.EVENT_TYPE_MODIFIED,
+        events.EVENT_TYPE_MOVED,
+    )
+
+    def dispatch(self, event: events.FileSystemEvent) -> None:
+        if event.event_type in self.EVENTS_WATCHED:
+            self.process_event(event)
+
+    @cached_property
+    def _extensions_to_watch(self) -> List[str]:
+        return [".py", *self.compiler_manager.registered_compilers.keys()]
```

Comment:
I believe the `.sol` / `.vy` files should be here as well?
For some reason, not working when I make changes to files in `contracts/`.
The tests never re-trigger.

Reply:
Ok, it seems to be working for me now... hmm, strange!